src/hoc/CustomLink.tsx

```diff
@@ -42,13 +43,16 @@ export const CustomLink = ({
 
   const { pathname, token } = getUrl()
 
+  const [shouldPrefetch, setShouldPrefetch] = useState(false)
+
   const handleMouseEnter = useCallback(() => {
-    router.prefetch(`${pathname}?token=${token}`)
+    // router.prefetch(`${pathname}?token=${token}`)
+    setShouldPrefetch(true)
```

Comment:
the dependency of of useCallback should be pathname and token right?

Reply:
The useCallback dependency array should include all variables that the callback depends on. In this case, the getUrl function depends on `href` and not directly on `pathname` and `token`. The `pathname` and `token` are derived from `href` within the getUrl function. Therefore, it is appropriate to use `href` in the dependency array. Including `pathname` and `token` would not be necessary since they are not direct dependencies but rather derived values.